tools/SourceKit/tools/sourcekitd/lib/API/sourcekitdAPI-InProc.cpp

```diff
@@ -0,0 +1,941 @@
+                                             sourcekitd_uid_t key,
+                                             int64_t val) {
+  static_cast<SKDObject *>(dict)->set(key, new SKDInt64(val));
+}
+
+void
+sourcekitd_request_dictionary_set_uid(sourcekitd_object_t dict,
+                                      sourcekitd_uid_t key,
+                                      sourcekitd_uid_t uid) {
+  static_cast<SKDObject *>(dict)->set(key, new SKDUID(uid));
+}
+
+sourcekitd_object_t
+sourcekitd_request_array_create(const sourcekitd_object_t *objects,
+                                size_t count) {
+  SKDArray *Array = new SKDArray();
+  for (size_t i = 0; i < count; ++i) {
+    Array->set(SOURCEKITD_ARRAY_APPEND, static_cast<SKDObject *>(objects[i]));
+  }
+  return retained(Array);
+}
+
+void
+sourcekitd_request_array_set_value(sourcekitd_object_t array, size_t index,
+                                   sourcekitd_object_t value) {
+  static_cast<SKDObject *>(array)->set(index, static_cast<SKDObject *>(value));
+}
+
+void
+sourcekitd_request_array_set_string(sourcekitd_object_t array, size_t index,
+                                    const char *string) {
+  static_cast<SKDObject *>(array)->set(index, 
+                                       new SKDString(std::string(string)));
+}
+
+void
+sourcekitd_request_array_set_stringbuf(sourcekitd_object_t array, size_t index,
+                                       const char *buf, size_t length) {
+  static_cast<SKDObject *>(array)->set(index,
+                                       new SKDString(std::string(buf, length)));
+}
+
+void
+sourcekitd_request_array_set_int64(sourcekitd_object_t array, size_t index,
+                                   int64_t val) {
+  static_cast<SKDObject *>(array)->set(index, new SKDInt64(val));
+}
+
+void
+sourcekitd_request_array_set_uid(sourcekitd_object_t array, size_t index,
+                                 sourcekitd_uid_t uid) {
+  static_cast<SKDObject *>(array)->set(index, new SKDUID(uid));
+}
+
+
+sourcekitd_object_t
+sourcekitd_request_int64_create(int64_t val) {
+  return retained(new SKDInt64(val));
+}
+
+sourcekitd_object_t
+sourcekitd_request_string_create(const char *string) {
+  return retained(new SKDString(std::string(string)));
+}
+
+sourcekitd_object_t
+sourcekitd_request_uid_create(sourcekitd_uid_t uid) {
+  return retained(new SKDUID(uid));
+}
+
+//===----------------------------------------------------------------------===//
+// Public Response API
+//===----------------------------------------------------------------------===//
+
+void
+sourcekitd_response_dispose(sourcekitd_response_t obj) {
+  static_cast<SKDObject *>(obj)->Release();
+}
+
+bool
+sourcekitd_response_is_error(sourcekitd_response_t obj) {
+  return isa<SKDError>(static_cast<SKDObject *>(obj));
+}
+
+sourcekitd_error_t
+sourcekitd_response_error_get_kind(sourcekitd_response_t obj) {
+  if (SKDError *Error = dyn_cast<SKDError>(static_cast<SKDObject *>(obj)))
```

Comment:
FYI, in such code patterns, you could use `if (auto *Error = dyn_cast<SDKError>(...))` which allows you to avoid adding the type name twice.